pkg/utils/informer/informermanager.go

```diff
@@ -255,6 +205,50 @@ func (s *informerManagerImpl) Stop() {
 	s.cancel()
 }
 
+func (s *informerManagerImpl) AddEventHandlerToInformer(resource schema.GroupVersionResource, handler cache.ResourceEventHandler) {
+	s.resourcesLock.Lock()
+	defer s.resourcesLock.Unlock()
+
+	// Get or create the informer (this is idempotent - if it exists, we get the same instance)
+	// The idempotent behavior is important because this method may be called multiple times,
+	// potentially concurrently, and relies on the shared informer instance from the factory.
+	informer := s.informerFactory.ForResource(resource).Informer()
```

Comment:
we did a transform on the informer cache in the other function but not here. Is it enough to do it once?